File: app/src/main/java/fr/iutlens/mmi/rogue/Sprite.java
package fr.iutlens.mmi.rogue;

/**
 * Created by dubois on 30/12/2019.
 */

class Sprite {
    public float prob;
    protected int id;
    int x,y;

    public Sprite(int tileId, int x, int y) {
        this.id = tileId;
        this.x = x;
        this.y = y;
        this.prob = 1;
    }

    public Sprite(int id, int x, int y, float prob) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.prob = prob;
    }

    public Sprite clone(){
        return new Sprite(id,x,y,prob);
    }

    public boolean getProb(){
        if (prob == 1) return true;
        return Math.random()<= prob;
    }

    int getId(){
        return id;
    }

    boolean block(){
        return Tile.get(id).hasOneFlag(Tile.F_BLOCK);
    }

    public void effect(Level level, Hero hero) {
        if (Tile.get(id).hasOneFlag(Tile.F_MONSTER|Tile.F_CONSOMMABLE)){
            level.removeContent(x,y);
        }
    }
}

File: app/src/main/java/fr/iutlens/mmi/rogue/Sprite.kt
package fr.iutlens.mmi.rogue

/**
 * Created by dubois on 30/12/2019.
 */
class Sprite {
    var prob: Float
    var id: Int
    var x: Int
    var y: Int

    constructor(tileId: Int, x: Int, y: Int) {
        id = tileId
        this.x = x
        this.y = y
        prob = 1f
    }

    constructor(id: Int, x: Int, y: Int, prob: Float) {
        this.id = id
        this.x = x
        this.y = y
        this.prob = prob
    }

    fun clone(): Sprite {
        return Sprite(id, x, y, prob)
    }

    fun getProb(): Boolean {
        return if (prob == 1f) true else Math.random() <= prob
    }

    fun block(): Boolean {
        return Tile[id]!!.hasOneFlag(Tile.F_BLOCK)
    }

    fun effect(level: Level?, hero: Hero?) {
        if (Tile[id]!!.hasOneFlag(Tile.F_MONSTER or Tile.F_CONSOMMABLE)) {
            level!!.removeContent(x, y)
        }
    }
}